model/load_balancer.rb

```diff
@@ -53,4 +55,14 @@ def remove_vm(vm)
   def hostname
     "#{name}.#{private_subnet.ubid[-5...]}.#{Config.load_balancer_service_hostname}"
   end
+
+  def need_certificates?
+    return true if certs_dataset.empty?
+
+    certs.all? { |lbc| lbc.created_at && lbc.created_at < Time.now - 60 * 60 * 24 * 30 * 2 }
```

Comment:
`created_at` column can't be null. So we don't need check `lbc.created_at` check

```ruby
column :created_at, :timestamp, null: false, default: Sequel.function(:now)
```

Reply:
also, it would crash, right?

Though, this raises a question...why is this code toggling between using the dataset, and then using the orm array? It seems like this could all be one SQL query. Also any time I see this kind of mixed code, I have to think carefully about how it interacts with an already-cached `certs` array that may have read a different snapshot. That can be okay when there's a nice efficiency or clarity advantage to mixing and matching, but I think this is an anti-example.


You can probably do some variant of `certs_dataset.where { created_at < Time.now - 60 * 60 * 24 * 30 * 2 }.empty?` unless you are carefully loading a snapshot of the data into `certs` for some reason.

Reply:
Updated